### DISCUSSION TITLE
Is Jint 3 beta ready for production use?

### DISCUSSION THREAD

**Discussion by david-w-lee** [sebastienros/jint#1473] [Q&A] [answered]
Our company has been using Jint 1 and we are trying to replace it with Jint 3.
Jint 3 has been around since 2016 but I believe it still stays in beta stage because ES6 hasn't been fully supported.
If we restrict its use to JavaScript 3, would you say it's stable enough to be used in production?

**Comment by lahma** [comment, marked as answer]
First and foremost, v3 is way faster than its predecessors, has less bugs, more features and not only conforms to wide test262 test suite but also Jints own extensive test suite.

Yes, beta comes mostly now because Jint is not feature complete and some cases like async/await work only partially, there are cases where there are still some problems. Public API should be quite stable, which also is a reason to have beta label to allow some wiggle room.

There are projects using Jint 3.x in production, like [RavenDB](https://ravendb.net/), [OrchardCore](https://orchardcore.net//)  and and [EventStore](https://www.eventstore.com/). They are quite big projects with demanding tasks and should give some pointers about whether Jint is ready for production.

If you enjoy Jint and get financial benefit out of it, don't hesitate to sponsor the project!

**Comment by david-w-lee** [comment]
Got it! Thanks.
